packages/integration-tests/src/process-controller/process_controller.ts

```diff
@@ -159,3 +164,75 @@ export const cdkCli = (
     env: options?.env,
   });
 };
+
+/**
+ *  packageManagerExecute - Factory function that returns a ProcessController for the specified package manager
+ */
+export const packageManagerExecute = (
+  packageManager: PackageManager,
+  args: string[] = [],
+  dir: string,
+  options?: {
+    env?: Record<string, string>;
+  }
+): ProcessController => {
+  const packageManagerExecutable = packageManager.startsWith('yarn')
+    ? 'yarn'
+    : packageManager;
+  const command = execaSync('npx', ['which', packageManagerExecutable], {
+    cwd: dir,
+  }).stdout.trim();
+  if (!command) {
+    throw new Error('Unable to locate packageManager binary');
+  }
+
+  let packageManagerBinary: PackageManagerExecutable;
+  switch (packageManager) {
+    case 'npm':
+      packageManagerBinary = 'npx';
+      break;
+
+    case 'pnpm':
+      packageManagerBinary = 'pnpm';
+      break;
+
+    case 'yarn-classic':
+    case 'yarn-modern':
+      packageManagerBinary = 'yarn';
+      break;
+
+    default:
+      throw new Error(`Unknown package manager: ${packageManager as string}`);
+  }
+  return new ProcessController(packageManagerBinary, args, {
+    cwd: dir,
+    env: options?.env,
+  });
+};
+
+/**
+ * packageManagerCli - Factory function that returns a ProcessController for the specified package manager
+ */
+export const packageManagerCli = (
```

Comment:
```suggestion
export const runPackageManager = (
```

Reply:
`packageManagerCli` is consistent with `amplifyCLI` and `cdkCli` though

Reply:
done d33f115b9e